React 18 Upgrade

ggdouglas opened the pull request:
This feature branch PR serves as the central hub for upgrading Blueprint's documentation and internal dependencies to **React 18**. The initial changes in this PR involve updating dependencies relevant to the React 18 upgrade. Subsequent changes will be made in separate PRs that will be be merged into `develop` and rebased into this feature branch.

**Tracking progress in the list below:**

##### Fixes type, lint, and compile breaks:
- [x] #7147
- [x] #7150
- [x] #7145
- [x] ~#7146~
- [x] #7173
- [x] #7174
- [x] #7161
- [x] #7159
- [x] #7160
- [x] #7162
- [x] #7177

##### Fixes/skips test breaks
- [x] #7148
- [x] #7169
- [x] #7170
- [x] #7171

##### Related issues:
- #7165
- #7166
- #7167
- #7168

ggdouglas committed:
Update React to 18.3.1 in docs/demo packages

ggdouglas committed:
Update to React 18.3.1 in other packages

ggdouglas committed:
Update React in test-commons and use enzyme-adapter-react-18

ggdouglas committed:
Update react-test-renderer to 18.3.1

ggdouglas committed:
Upgrade @testing-library/react to latest

ggdouglas committed:
Update `@types/react` and `@types/react-dom` at root

ggdouglas committed:
Remove deprecated ReactDOM methods from table-dev-app

ggdouglas committed:
Replace deprecated ReactDOM methods in toastCreateAsyncExample

ggdouglas committed:
Remove deprecated ReactDOM methods from demo-app

svc-palantir-github commented:
<h3>Remove deprecated ReactDOM methods from demo-app</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/b28d5773-ddd0-4800-8709-ca9d0b5b3d67/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/b28d5773-ddd0-4800-8709-ca9d0b5b3d67/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/b28d5773-ddd0-4800-8709-ca9d0b5b3d67/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/b28d5773-ddd0-4800-8709-ca9d0b5b3d67/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

ggdouglas committed:
Replace deprecated ReactDOM methods for docs-app

svc-palantir-github commented:
<h3>Replace deprecated ReactDOM methods for docs-app</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/e725be90-ddf1-4aa0-b257-414060edc1a5/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/e725be90-ddf1-4aa0-b257-414060edc1a5/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/e725be90-ddf1-4aa0-b257-414060edc1a5/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/e725be90-ddf1-4aa0-b257-414060edc1a5/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>Replace deprecated ReactDOM methods for docs-app</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/5dbb5cca-0c16-4063-b3d6-389a6c3e6d2b/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/5dbb5cca-0c16-4063-b3d6-389a6c3e6d2b/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/5dbb5cca-0c16-4063-b3d6-389a6c3e6d2b/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/5dbb5cca-0c16-4063-b3d6-389a6c3e6d2b/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>Replace ResizeHandle with React.JSX.Element in resizableTest</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/a56ee424-6980-431e-8b93-7faaa840ccae/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/a56ee424-6980-431e-8b93-7faaa840ccae/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/a56ee424-6980-431e-8b93-7faaa840ccae/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/a56ee424-6980-431e-8b93-7faaa840ccae/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>Replace ResizeHandle with React.JSX.Element in resizableTest</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/c0edadbe-1931-42df-9146-93f09d03f405/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/c0edadbe-1931-42df-9146-93f09d03f405/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/c0edadbe-1931-42df-9146-93f09d03f405/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/c0edadbe-1931-42df-9146-93f09d03f405/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

ggdouglas committed:
Replace deprecated act from react-dom/test-utils with export from react

Fixes the following lint error on `TestUtils.act`:
```
The signature '(callback: () => VoidOrUndefinedOnly): void' of 'TestUtils.act' is deprecated.
```

svc-palantir-github commented:
<h3>Disable ReactDOM API lint deprecations in contextMenuSingleton</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/d288243e-2301-4590-aeb4-90c8c6d97009/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/d288243e-2301-4590-aeb4-90c8c6d97009/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/d288243e-2301-4590-aeb4-90c8c6d97009/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/d288243e-2301-4590-aeb4-90c8c6d97009/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>(temp) Skip failing tests to check for stability/test flake</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/462dabc7-ba84-49b3-ad21-53a8099de374/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/462dabc7-ba84-49b3-ad21-53a8099de374/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/462dabc7-ba84-49b3-ad21-53a8099de374/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/462dabc7-ba84-49b3-ad21-53a8099de374/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>(temp) Skip more tests that are now failing in CI</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/42dd4f5e-2498-4132-ac4e-567114797964/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/42dd4f5e-2498-4132-ac4e-567114797964/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/42dd4f5e-2498-4132-ac4e-567114797964/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/42dd4f5e-2498-4132-ac4e-567114797964/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>Ignore deprecated ReactDOM methods in OverlayToaster</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/f69b40c7-014d-4ae6-b457-a10f050c92f7/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/f69b40c7-014d-4ae6-b457-a10f050c92f7/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/f69b40c7-014d-4ae6-b457-a10f050c92f7/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/f69b40c7-014d-4ae6-b457-a10f050c92f7/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>Skip additional test flake</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/dcf4221b-576b-4502-a548-f6a2828c74d5/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/dcf4221b-576b-4502-a548-f6a2828c74d5/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/dcf4221b-576b-4502-a548-f6a2828c74d5/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/dcf4221b-576b-4502-a548-f6a2828c74d5/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>Skip flaky tests</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/ef8896ef-38d0-4900-b748-9110c1f9ee2b/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/ef8896ef-38d0-4900-b748-9110c1f9ee2b/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/ef8896ef-38d0-4900-b748-9110c1f9ee2b/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/ef8896ef-38d0-4900-b748-9110c1f9ee2b/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>Skip failing tests resulting from React 18 upgrade</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/6a8f9907-6a05-4119-adff-cc4c02eb2c12/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/6a8f9907-6a05-4119-adff-cc4c02eb2c12/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/6a8f9907-6a05-4119-adff-cc4c02eb2c12/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/6a8f9907-6a05-4119-adff-cc4c02eb2c12/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

ggdouglas commented on packages/core/test/controls/numericInputTests.tsx:
```diff
@@ -1098,7 +1097,7 @@ describe("<NumericInput>", () => {
             incrementButton.simulate("mousedown", { shiftKey: true });
             expect(component.find("input").prop("value")).to.equal("1.101");
 
-            TestUtils.act(() => {
```
Fixes a deprecation in React 18. Act is now exported from React as of `v18.3.1`.

![Screenshot 2025-01-14 at 17 12 11@2x](https://github.com/user-attachments/assets/fd04a92b-c121-4109-88a4-c03f7ec6a641)

https://react.dev/warnings/react-dom-test-utils#reactdomtestutilsact-warning

> act from react-dom/test-utils has been deprecated in favor of act from react.

ggdouglas commented on packages/core/src/components/icon/icon.tsx:
```diff
@@ -101,7 +101,7 @@ export interface DefaultIconProps extends IntentProps, Props, DefaultSVGIconProp
  * @see https://stackoverflow.com/a/73795494/7406866
  */
 export interface IconComponent extends React.FC<IconProps<Element>> {
-    <T extends Element = Element>(props: IconProps<T>): React.ReactElement | null;
```
Attempted to cherry-pick this type change into develop, but unfortunately encountered errors. This type fix is valid in React 18, but fails with React 16 types.

See type failures on branch based off of develop: https://github.com/palantir/blueprint/compare/gdouglas/test-Icon-compile-break

gluxon replied:
I know I suggested the `React.ReactElement` to `React.ReactNode` change, but I'm realizing we might need something more dynamic to maintain both React 16 and 18 compatibility.

We could try this:

```ts
<T extends Element = Element>(props: IconProps<T>): ReturnType<typeof React.FC< IconProps<Element>>> | null;
```

I think that should return `React.ReactNode` on `@types/react@18` and `React.ReactElement` on `@types/react@16`?

When Blueprint drops React 16/17 compatibility, the `ReturnType` hack above could be replaced with `React.ReactNode` again.

ggdouglas replied:
I think this will work! Thanks for the suggestion. #7177

ggdouglas commented on packages/icons/src/svgIconContainer.tsx:
```diff
@@ -42,7 +42,7 @@ export type SVGIconContainerProps<T extends Element> = Omit<SVGIconProps<T>, "ch
  * @see https://stackoverflow.com/a/73795494/7406866
  */
 export interface SVGIconContainerComponent extends React.FC<SVGIconContainerProps<Element>> {
-    <T extends Element = Element>(props: SVGIconContainerProps<T>): React.ReactElement | null;
```
Similar to [type compile issue with Icon](https://github.com/palantir/blueprint/pull/7142#discussion_r1915691062), this fix compiles in React 18, but not 16.

See: https://github.com/palantir/blueprint/compare/gdouglas/test-SVGIconContainer-compile-break

svc-palantir-github commented:
<h3>Patch react-day-picker@7.4.9 in datetime devDependencies</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/00e3329e-f7b8-46e1-9d4a-172f8a6096a5/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/00e3329e-f7b8-46e1-9d4a-172f8a6096a5/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/00e3329e-f7b8-46e1-9d4a-172f8a6096a5/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/00e3329e-f7b8-46e1-9d4a-172f8a6096a5/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>Run yarn dedupe</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/77493a0c-8bd0-43ac-b9c3-109bca4e1cc2/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/77493a0c-8bd0-43ac-b9c3-109bca4e1cc2/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/77493a0c-8bd0-43ac-b9c3-109bca4e1cc2/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/77493a0c-8bd0-43ac-b9c3-109bca4e1cc2/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>Run yarn dedupe</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/7014de9f-bc5a-4a53-9cec-5cd92d72fc33/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/7014de9f-bc5a-4a53-9cec-5cd92d72fc33/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/7014de9f-bc5a-4a53-9cec-5cd92d72fc33/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/7014de9f-bc5a-4a53-9cec-5cd92d72fc33/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

ggdouglas committed:
Skip test that passes locally

svc-palantir-github commented:
<h3>Skip test that passes locally</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/d3960006-26c4-4fb3-8b41-bdff17df7d4a/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/d3960006-26c4-4fb3-8b41-bdff17df7d4a/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/d3960006-26c4-4fb3-8b41-bdff17df7d4a/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/d3960006-26c4-4fb3-8b41-bdff17df7d4a/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

ggdouglas commented on packages/datetime/package.json:
```diff
@@ -68,10 +68,10 @@
         "karma": "^6.4.2",
         "mocha": "^10.2.0",
         "npm-run-all": "^4.1.5",
-        "react": "^16.14.0",
```
Should we be updating the react `devDependencies` in the datetime package to 18 if it has `peerDependencies` of 16/17?

Wondering if I should leave these be instead of changing them.

ggdouglas committed:
Run yarn dedupe

svc-palantir-github commented:
<h3>Run yarn dedupe</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/b3e4e4f8-1070-44ad-8190-b566d6eb697a/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/b3e4e4f8-1070-44ad-8190-b566d6eb697a/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/b3e4e4f8-1070-44ad-8190-b566d6eb697a/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/b3e4e4f8-1070-44ad-8190-b566d6eb697a/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

ggdouglas commented on packages/test-commons/bootstrap.mjs:
```diff
@@ -2,8 +2,10 @@
  * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
  */
 
+import ReactEighteenAdapter from "@cfaester/enzyme-adapter-react-18";
```
https://github.com/cfaester/enzyme-adapter-react-18
> "A **very** unofficial adapter for React 18 for [Enzyme](https://enzymejs.github.io/enzyme/)."

Very unofficial, but it at least gets us moving forward until we can replace Enzyme.

gluxon commented on packages/docs-app/src/examples/core-examples/toastCreateAsyncExample.tsx:
```diff
@@ -88,12 +88,13 @@ async function showMessageFromNewToaster() {
         throw new Error("No elements were found under Toaster container.");
     }
-    ReactDOM.unmountComponentAtNode(toasterRenderRoot);
+    const root = ReactDOM.createRoot(toasterRenderRoot);
```
I believe this calls `ReactDOM.createRoot` on a DOM node that previously had a react root created on it, which leaks memory. The old root never gets unmounted.

The `OverlayToaster.createAsync` call above doesn't seem to override the `domRenderer` field. I think the React root that was previously created on `toasterRenderRoot` is actually a React 17 legacy root. Without changes to how `createAsync` is called above, I think `ReactDOM.unmountComponentAtNode` is most correct here?

To fully remove `unmountComponentAtNode`, there would need to be a change to the `createAsync` call to create a React root, store a reference to it in the example here, and then call unmount in the cleanup effect.

ggdouglas replied:
Ah, yup, good catch! I'll split this out, ignore the deprecated ReactDOM lint errors, and track this in #7166

ggdouglas replied:
595f50be28e84d6845240a90416855faef095c65 / 89ce662ec2694accab04244454a9225bc0b527e5

ggdouglas committed:
Revert change to ReactDOM method in toastCreateAsyncExample

ggdouglas committed:
Ignore deprecated ReactDOM method in toastCreateAsyncExample

svc-palantir-github commented:
<h3>Ignore deprecated ReactDOM method in toastCreateAsyncExample</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/6d3f0a58-e2c9-43c4-b75b-d97e6d6ed08c/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/6d3f0a58-e2c9-43c4-b75b-d97e6d6ed08c/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/6d3f0a58-e2c9-43c4-b75b-d97e6d6ed08c/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/6d3f0a58-e2c9-43c4-b75b-d97e6d6ed08c/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>Update return types for Icon and SVGIconContainer</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/cc3a336e-6b39-403b-8eb6-c9ba3d9a9193/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/cc3a336e-6b39-403b-8eb6-c9ba3d9a9193/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/cc3a336e-6b39-403b-8eb6-c9ba3d9a9193/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/cc3a336e-6b39-403b-8eb6-c9ba3d9a9193/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

svc-palantir-github commented:
<h3>Ignore deprecated ReactDOM method in toastCreateAsyncExample</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/8c6a4ed8-311c-4e87-b886-95fbd1ca1689/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/8c6a4ed8-311c-4e87-b886-95fbd1ca1689/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/8c6a4ed8-311c-4e87-b886-95fbd1ca1689/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/8c6a4ed8-311c-4e87-b886-95fbd1ca1689/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

ggdouglas committed:
Skip tests that pass locally

svc-palantir-github commented:
<h3>Skip tests that pass locally</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/4d52832c-79cd-4f64-8e4e-9fc6e5a77e9c/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/4d52832c-79cd-4f64-8e4e-9fc6e5a77e9c/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/4d52832c-79cd-4f64-8e4e-9fc6e5a77e9c/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/4d52832c-79cd-4f64-8e4e-9fc6e5a77e9c/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>

ggdouglas committed:
Skip tests that pass locally

svc-palantir-github commented:
<h3>Skip tests that pass locally</h3>
Build artifact links for this commit: <strong><a href="https://output.circle-artifacts.com/output/job/1814d8d1-81f9-4551-a775-5ede1845922e/artifacts/0/packages/docs-app/dist/index.html">documentation</a> | <a href="https://output.circle-artifacts.com/output/job/1814d8d1-81f9-4551-a775-5ede1845922e/artifacts/0/packages/landing-app/dist/index.html">landing</a> | <a href="https://output.circle-artifacts.com/output/job/1814d8d1-81f9-4551-a775-5ede1845922e/artifacts/0/packages/table-dev-app/dist/index.html">table</a> | <a href="https://output.circle-artifacts.com/output/job/1814d8d1-81f9-4551-a775-5ede1845922e/artifacts/0/packages/demo-app/dist/index.html">demo</a></strong>

<em>This is an automated comment from the deploy-preview CircleCI job.</em>
